server/api.js

```diff
@@ -50,22 +50,36 @@ router.get("/energizers/:id", async (req, res) => {
 	}
 });
 
-//I have only created this POST endpoint to update my database, its very incomplete as it lacks any validation, but it works and can be used as a base
-
-router.post("/energizers", (req, res) => {
-	let name = req.query.name;
-	let description = req.query.description;
-
-	const insertQuery = "INSERT INTO energizers(name, description) VALUES($1, $2)";
-	db.query(insertQuery, [name, description])
-		.then(() => {
-			res
-				.status(201)
-				.json({ message: "The energizer was succesfully uploaded" });
-		})
-		.catch((error) => {
-			res.status(500).send(error);
-		});
+
+
+
+router.post("/energizers", async (req, res) => {
+	const { name, description } = req.body;
+
+	if (!name || !description) {
+		return res.status(400).json({ error: "Missing required fields" });
+	}
+    if(name.length > 50){
+        return res.status(400).json({ error: "Name has to be less or equal to 50 characters" });
+    }
+
+	try {
+		await db.query(
+			"INSERT INTO energizers( name, description) VALUES ($1, $2)",
+			[ name, description]
+		);
+		const result = await db.query(`SELECT energizers.id, energizers.name, energizers.description, ROUND(AVG(energizer_ratings.rating), 1)  AS rating
+        FROM  energizers 
+        LEFT JOIN energizer_ratings ON energizers.id = energizer_ratings.energizer_id GROUP BY energizers.id`);
+        res.json(result.rows);
```

Comment:
I don't think it's good behaviour for a POST request, but you want this behaviour I think this code should be re-used from the GET request (by placing it in a separate shared function that both API calls can use) so that they are always kept up to date together.